tooling/internal/cpp/binary/binary.go

```diff
@@ -84,18 +84,18 @@ func writeHeaderFile(env *dsl.Environment, options packaging.CppCodegenOptions)
 			fmt.Fprintf(w, "class %s : public %s, yardl::binary::BinaryWriter {\n", writerClassName, common.QualifiedAbstractWriterName(protocol))
 			w.Indented(func() {
 				w.WriteStringln("public:")
-				fmt.Fprintf(w, "%s(std::ostream& stream)\n", writerClassName)
+				fmt.Fprintf(w, "%s(std::ostream& stream, Version version = Version::Latest)\n", writerClassName)
```

Comment:
The name `Latest` could be misleading. Imagine you have a an existing codebase written against v1. Then v2 comes along and you just want to reference it to genereate the ability to write to it. In that case v1 is more "current" than "latest"